simtools/corsika/corsika_output.py

```diff
@@ -0,0 +1,1002 @@
+            Label to identify to which histogram the new axis belongs.
+
+        Raises
+        ------
+        ValueError:
+            if label is not valid.
+        """
+        transform = {"log": bh.axis.transform.log, "linear": None}
+
+        if label not in self._allowed_histograms:
+            msg = "allowed labels must be one of the following: {}".format(self._allowed_histograms)
+            self._logger.error(msg)
+            raise (ValueError)
+
+        all_axes = ["X axis", "Y axis"]
+        if label == "hist_position":
+            all_axes.append("Z axis")
+
+        boost_axes = []
+        for axis in all_axes:
+            try:
+                boost_axes.append(
+                    bh.axis.Regular(
+                        bins=self.hist_config[label][axis]["bins"],
+                        start=self.hist_config[label][axis]["start"].value,
+                        stop=self.hist_config[label][axis]["stop"].value,
+                        transform=transform[self.hist_config[label][axis]["scale"]],
+                    )
+                )
+            except AttributeError:
+                boost_axes.append(
+                    bh.axis.Regular(
+                        bins=self.hist_config[label][axis]["bins"],
+                        start=self.hist_config[label][axis]["start"],
+                        stop=self.hist_config[label][axis]["stop"],
+                        transform=transform[self.hist_config[label][axis]["scale"]],
+                    )
+                )
+        return boost_axes
+
+    def _create_histograms(self):
+        """
+        Create the histogram instances.
+        """
+        if self.telescope_indices is None:
+            self.num_of_hist = 1
+        else:
+            self.num_of_hist = len(self.telescope_indices)
+
+        self.hist_position, self.hist_direction, self.hist_time_altitude = [], [], []
+
+        for step in range(self.num_of_hist):
+            boost_axes_position = self._create_regular_axes("hist_position")
+            self.hist_position.append(
+                bh.Histogram(
+                    boost_axes_position[0],
+                    boost_axes_position[1],
+                    boost_axes_position[2],
+                )
+            )
+            boost_axes_direction = self._create_regular_axes("hist_direction")
+            self.hist_direction.append(
+                bh.Histogram(
+                    boost_axes_direction[0],
+                    boost_axes_direction[1],
+                )
+            )
+
+            boost_axes_time_altitude = self._create_regular_axes("hist_time_altitude")
+            self.hist_time_altitude.append(
+                bh.Histogram(
+                    boost_axes_time_altitude[0],
+                    boost_axes_time_altitude[1],
+                )
+            )
+
+    def _fill_histograms(self, photons, azimuth_angle=None, zenith_angle=None):
+        """Fill the histograms created by self._create_histogram
```

Comment:
Indeed, it is multi-purpose. It fills all the histograms (we cant pick only one) and it works either for all telescopes together or single telescopes, depending on the flag. I did not feel the need to include more explanation because it is an internal function. Regarding what it feels with (Cherenkov photons), I thought it was clear from the purpose of the class, but I added now a sentence to clarify it.

Reply:
You are right that generally internal functions don't need to be super documented. However, in cases of convoluted code, the documentation is helpful for other developers/reviewers to understand it.